src/foam/nanos/auth/Permission.js

```diff
@@ -23,17 +23,10 @@
    ],
 
    methods: [
-     function implies(permission) {
-       if ( this.id.endsWith('*') ) {
-         var prefix = this.id.substring(0, this.id.length-1);
+     function implies(gPermissionId, permissionId) {
+         var prefix = gPermissionId.substring(0, gPermissionId.length-1);
 
-         // Check that is a valid permission string (should be done as a property validator)
-         if ( prefix.length && ! prefix.endsWith('.') ) return false;
-
-         return permission.startsWith(prefix);
-       }
-
-       return permission == this.id;
+         return permissionId.startsWith(prefix);
```

Comment:
This check is too simple. Suppose you own account #1 so I give you permission account.read.1 so you can read your account balance. But consider if I have account 10. Just doing startsWith() would give your permission to read my account also, since account.read.10 startsWith account.read.1.